core/arch/arm/mm/core_mmu_lpae.c

```diff
@@ -170,7 +170,10 @@
 
 
 
+#ifndef ADDR_SPACE_SIZE
```

Comment:
Since you're making this configurable externally, I'd rather make it a proper `CFG_` variable.
If you add `CFG_LPAE_ADDR_SIZE ?= (1ull << 32)` in `mk/config.mk` and rename ADDR_SPACE_SIZE TO CFG_LPAE_ADDR_SIZE in this file, it should work.